src/sparse_blas/backends/cusparse/operations/cusparse_spmm.cpp

```diff
@@ -0,0 +1,286 @@
+        auto cu_op_a = get_cuda_operation(type, opA);
+        auto cu_op_b = get_cuda_operation(type, opB);
+        auto cu_type = get_cuda_value_type(type);
+        auto cu_alg = get_cuda_spmm_alg(alg);
+        set_pointer_mode(cu_handle, is_alpha_host_accessible);
+        auto status = cusparseSpMM_bufferSize(cu_handle, cu_op_a, cu_op_b, alpha, cu_a, cu_b, beta,
+                                              cu_c, cu_type, cu_alg, &temp_buffer_size);
+        check_status(status, __func__);
+    };
+    auto event = dispatch_submit(__func__, queue, functor, A_handle, B_handle, C_handle);
+    event.wait_and_throw();
+    spmm_descr->temp_buffer_size = temp_buffer_size;
+    spmm_descr->buffer_size_called = true;
+}
+
+inline void common_spmm_optimize(oneapi::mkl::transpose opA, oneapi::mkl::transpose opB,
+                                 bool is_alpha_host_accessible, matrix_view A_view,
+                                 matrix_handle_t A_handle, dense_matrix_handle_t B_handle,
+                                 bool is_beta_host_accessible, dense_matrix_handle_t C_handle,
+                                 spmm_alg alg, spmm_descr_t spmm_descr) {
+    check_valid_spmm("spmm_optimize", opA, opB, A_view, A_handle, B_handle, C_handle,
+                     is_alpha_host_accessible, is_beta_host_accessible, alg);
+    if (!spmm_descr->buffer_size_called) {
+        throw mkl::uninitialized("sparse_blas", "spmm_optimize",
+                                 "spmm_buffer_size must be called before spmm_optimize.");
+    }
+    spmm_descr->optimized_called = true;
+    spmm_descr->last_optimized_opA = opA;
+    spmm_descr->last_optimized_opB = opB;
+    spmm_descr->last_optimized_A_view = A_view;
+    spmm_descr->last_optimized_A_handle = A_handle;
+    spmm_descr->last_optimized_B_handle = B_handle;
+    spmm_descr->last_optimized_C_handle = C_handle;
+    spmm_descr->last_optimized_alg = alg;
+}
+
+void spmm_optimize_impl(cusparseHandle_t cu_handle, oneapi::mkl::transpose opA,
+                        oneapi::mkl::transpose opB, const void* alpha, matrix_handle_t A_handle,
+                        dense_matrix_handle_t B_handle, const void* beta,
+                        dense_matrix_handle_t C_handle, spmm_alg alg, void* workspace_ptr,
+                        bool is_alpha_host_accessible) {
+    auto cu_a = A_handle->backend_handle;
+    auto cu_b = B_handle->backend_handle;
+    auto cu_c = C_handle->backend_handle;
+    auto type = A_handle->value_container.data_type;
+    auto cu_op_a = get_cuda_operation(type, opA);
+    auto cu_op_b = get_cuda_operation(type, opB);
+    auto cu_type = get_cuda_value_type(type);
+    auto cu_alg = get_cuda_spmm_alg(alg);
+    set_pointer_mode(cu_handle, is_alpha_host_accessible);
+    auto status = cusparseSpMM_preprocess(cu_handle, cu_op_a, cu_op_b, alpha, cu_a, cu_b, beta,
+                                          cu_c, cu_type, cu_alg, workspace_ptr);
+    check_status(status, "optimize_spmm");
```

Comment:
```suggestion
    check_status(status, "spmm_optimize");
```

Reply:
Fixed in https://github.com/oneapi-src/oneMKL/pull/527/commits/325f79495c2090b95538e0eaaa80ccf40f2e9737